packages/app/src/cli/services/app-logs/dev/write-app-logs.ts

```diff
@@ -30,11 +33,15 @@ export const writeAppLogsToFile = async ({
     const toSaveData = camelcaseKeys(
       {
         ...appLog,
-        payload: JSON.parse(appLog.payload),
+        payload: appLogPayload,
       },
       {deep: true},
     )
 
+    if (appLogPayload instanceof FunctionRunLog) {
+      toSaveData.payload.logs = appLogPayload.logs.split('\n').filter(Boolean)
+    }
```

Comment:
This is a bit annoying because of the typing.
The `FunctionRunLog`, as in the type we have for the data coming from the response, has the `logs` as a string.
But here, we want the logs split out for our write, so it needs to be a `string[]`.

However, because of the `payload: appLogPayload` on line 36, the `toSaveData.payload` is our generic AppLogPayload type, which doesn't have `payload.logs`, and even if it was our `FunctionRunLog` type, `string` wouldn't match `string[]`.

Right now it works because it is getting the `| any` from the method sig, but I could see a case for putting `payload: appLogPayload as any` on line 36 here.

Reply:
Could we reverse the order here and mutate `appLogPayload` itself before creating `toSaveData`?

Reply:
I dont think this changes much because:
- to look for logs to modify, we need `appLogPayload instanceof FunctionRunLog`
- but FunctionRunLog.logs is a `string`, so we get `Type 'string[]' is not assignable to type 'string'`, if we try to mutate the appLogPayload.logs to the split string.

I have it setup as is in order to preserve the `appLogPayload`, with the idea being that that object we're going to write (toSaveData) should be the one that can handle all the different ways of being written.

Reply:
That said, just putting `any` on the relevant field also seems fine. My opinion isn't super strong.

Both of those options are preferable to having the `any` in the type signature IMO.

Reply:
I like putting `any` on the `saveData` object, that makes the most sense to me, since we're just creating this object to be written, aka not to be reused anywhere outside of this.

I think the `any` on the type signature is a separate thing though? As in, the type should remain `AppLogPayload | any` imo because of https://github.com/Shopify/cli/pull/4255#discussion_r1700481759